src/ui/public/utils/subscribe.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IScope } from 'angular';
+import * as Rx from 'rxjs';
+import { fatalError } from 'ui/notify/fatal_error';
+
+function callInDigest<T extends any[]>($scope: IScope, fn: (...args: T) => void, ...args: T) {
+  try {
+    // this is terrible, but necessary to synchronously deliver subscription values
+    // to angular scopes. This is required by some APIs, like the `config` service,
+    // and beneficial for root level directives where additional digest cycles make
+    // kibana sluggish to load.
+    //
+    // If you copy this code elsewhere you better have a good reason :)
+    if ($scope.$root.$$phase) {
+      fn(...args);
+    } else {
+      $scope.$apply(() => fn(...args));
+    }
+  } catch (error) {
+    fatalError(error);
```

Comment:
__question:__ not sure if it's a big deal, but shouldn't we unsubscribe from original observable if we call `fatalError` when `next` fails?

Reply:
Rxjs takes care of that because `fatalError()` rethrows the error and when a `next()` handler throws the parent is unsubscribed.

Reply:
Ah, right, I forgot that `fatalError` always throws.